src/System.Numerics.Vectors/src/System/Numerics/Vector.cs

```diff
@@ -363,7 +363,8 @@ public unsafe Vector(T[] values, int index)
         {
             if (values == null)
             {
-                throw new ArgumentNullException("values");
+                // Match the JIT's exception type here. For perf, a NullReference is thrown instead of an ArgumentNull.
+                throw new NullReferenceException(SR.Arg_NullArgumentNullRef);
             }
             if (index < 0 || (values.Length - index) < Count)
```

Comment:
That would be an option, but I actually like at least giving the exception message here. It will help (a bit) with figuring out your error, whereas with the default exception, you might not immediately know what happened (should still be obvious based on which method is throwing it).

Also, since this is just the IL version, perf here doesn't really matter. This type will be pretty slow in general unless you are running with JIT intrinsics.